File: chess_cpp/validators/DestinationValidator.cpp
//
// Created by Hrach Khachatryan on 12/15/19.
//

#include "DestinationValidator.h"

bool DestinationValidator::validate(Board *board, Spot *start, Spot *end) {
    return end->getPiece() == nullptr || end->getPiece()->isWhite() == start->getPiece()->isWhite();
}

std::string DestinationValidator::getNotValidMessage() {
    return MessageHandler::getHellNoMessage();
}

File: chess_java/src/chess/validators/DestinationValidator.java
package chess.validators;

import chess.Board;
import chess.Spot;
import chess.utils.MessageHandler;

public class DestinationValidator extends PieceMoveValidator {


    @Override
    public boolean validate(Board board, Spot start, Spot end) {
        return end.getPiece() == null || end.getPiece().isWhite() == start.getPiece().isWhite();
    }

    @Override
    public String getNotValidMessage() {
        return MessageHandler.getHellNoMessage();
    }
}
